### PR TITLE
LPS-96805 Enable Headless Tests and fix those which are failing

### PR DESCRIPTION
**Pull request by ruben-pulido**

### CODE REVIEW

**Comment by ruben-pulido** [comment]
ci:test

**Comment by nhpatt** [comment]
Let's see... reviewing right now, sorry for the delay :S

File: modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/RESTBuilder.java [hunk 1]
```diff
@@ -134,6 +134,10 @@ public void build() throws Exception {
 
 		context.put("freeMarkerTool", freeMarkerTool);
 
+		OpenAPIParserUtil openAPIParserUtil = OpenAPIParserUtil.getInstance();
+
+		context.put("OpenAPIParserUtil", openAPIParserUtil);
```
Comment (nhpatt) [inline modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/RESTBuilder.java]: Don't expose OpenAPIParserUtil, use a new method in FreeMarkerTool.

File: modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/RESTBuilder.java [hunk 2]
```diff
@@ -232,7 +237,9 @@ public void build() throws Exception {
 
 				Schema schema = entry.getValue();
 
-				_putSchema(context, schema, schemaName);
+				_putSchema(
+					context, schema, schemaName,
+					_getSubentities(allSchemas, javaMethodSignatures));
```
Comment (nhpatt) [inline modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/RESTBuilder.java]: But subschemas can be exported to the context like allSchemas, but I guess they are scoped to a schema so it doesn't make sense

File: modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/RESTBuilder.java [hunk 3]
```diff
@@ -1322,8 +1329,36 @@ private String _fixOpenAPISchemaPropertyNames(
 		return operations;
 	}
 
+	private Set<String> _getSubentities(
+		Map<String, Schema> allSchemas,
```
Comment (nhpatt) [inline modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/RESTBuilder.java]: schemas

File: modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/RESTBuilder.java [hunk 4]
```diff
@@ -1336,6 +1371,8 @@ private void _putSchema(
 		context.put("schemaVarName", schemaVarName);
 		context.put(
 			"schemaVarNames", TextFormatter.formatPlural(schemaVarName));
+
+		context.put("subentities", subentities);
```
Comment (nhpatt) [inline modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/RESTBuilder.java]: Ok, I see it here 👍

File: modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/util/OpenAPIParserUtil.java [hunk 5]
```diff
@@ -328,6 +332,11 @@ public static boolean hasHTTPMethod(
 		return false;
 	}
 
+	private OpenAPIParserUtil() {
```
Comment (nhpatt) [inline modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/util/OpenAPIParserUtil.java]: better from FreeMarkerTool :)

**Pull request closed without merging**